### ISSUE TITLE
db: consider adapting LoadBlockSema to limit concurrent bytes loaded

### ISSUE THREAD

**Issue by jbowens** [cockroachdb/pebble#4185]
Memory for in-flight reads is allocated before a block can be inserted into the block cache, so these reads do not drawn down from the block cache's memory. A spike in concurrent block loads can cause a spike in memory allocations. The LoadBlockSema semaphore used to limit the _count_ of concurrent loads would probably be more effective as a limit on the concurrent bytes loaded. This might even be a better proxy for concurrent read i/os, since large reads may translate to many individual block i/os (especially as it pertains to cloud platforms concurrent iop budget calculations).

Jira issue: PEBBLE-311

**Comment by jbowens** [comment]
@RaduBerinde should we just close this out as not planned?

**Comment by RaduBerinde** [comment]
Yes. I was considering adding a byte limit per read shard in addition to the current semaphore; but I'm not sure we need it.
